server/queueMethods.go

```diff
@@ -167,7 +183,12 @@ func (channel *Channel) queueUnbind(method *amqp.QueueUnbind) *amqp.Error {
 		return err
 	}
 
-	bind := binding.NewBinding(method.Queue, method.Exchange, method.RoutingKey, method.Arguments, ex.ExType() == exchange.ExTypeTopic)
+	bind, err := binding.NewBinding(method.Queue, method.Exchange, method.RoutingKey, method.Arguments, ex.ExType() == exchange.ExTypeTopic)
+
+	if err != nil {
+
+	}
+
 	ex.RemoveBinding(bind)
```

Comment:
When we remove or add binding we check them for equal. Now it checks only exchange, queue and routing key. Is it right way for headers bindings?

Reply:
I'd need to double-check the specification to be sure, I'll keep you updated on that

Reply:
According to the specification:
```
1.7.2.5.5. Parameter queue.unbind.arguments (table)
Ordinal: 5
Domain: table
Label: arguments of binding
Specifies the arguments of the binding to unbind.
```

I'd think the remove binding should also have the matching arguments as well then, I'll update `RemoveBinding`

Reply:
Should be done now as well